crates/ruff_python_ast/src/nodes.rs

```diff
@@ -2578,13 +2577,36 @@ impl Ranged for Identifier {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Int {
```

Comment:
I like the new abstraction but I think we should hide its internal to make it easier to change the internals in the future (e.g. using tagged pointers). 

* Make `Int` a `struct` with one inner field
* Create a private enum that represents the two different states
* Implement `Eq`, `PartialEq` for all types for which it is safe (`i8..i64`, `u8..u32`).
* Implement `as_*` helpers that either return `T` if the conversion is safe or `Option<T>` if the value falls into the `> i64` range. 
* Implement `Debug` to hide the internal representation (should always serialize to the value)